src/test/util/event.rs

```diff
@@ -99,17 +125,64 @@ impl std::ops::DerefMut for EventClient {
 
 impl EventClient {
     pub async fn new() -> Self {
+        EventClient::with_options(None).await
+    }
+
+    pub async fn with_options(options: impl Into<Option<ClientOptions>>) -> Self {
         let handler = EventHandler::default();
         let command_events = handler.command_events.clone();
         let pool_cleared_events = handler.pool_cleared_events.clone();
-        let client = TestClient::with_handler(Some(handler)).await;
+        let client = TestClient::with_handler(Some(handler), options).await;
+
+        // clear events from commands used to set up client.
+        command_events.write().unwrap().clear();
 
         Self {
             client,
             command_events,
             pool_cleared_events,
         }
     }
+
+    /// Gets the first started/succeeded pair of events for the given command name, popping off all
+    /// events before and between them.
+    ///
+    /// Panics if the command failed or could not be found in the events.
+    pub fn get_successful_command_execution(
+        &self,
+        command_name: &str,
+    ) -> (CommandStartedEvent, CommandSucceededEvent) {
+        let mut command_events = self.command_events.write().unwrap();
+
+        let mut started: Option<CommandStartedEvent> = None;
+
+        while let Some(event) = command_events.pop_front() {
+            if event.command_name() == command_name {
+                match started {
+                    None => {
```

Comment:
Would it make sense to just move this branch above the loop? It seems like it will always run in the first iteration and never again after.

Reply:
It needs to be in a loop in case there are events at the beginning of the queue which aren't associated with the given command name.

Reply:
Ah, I missed that the match was inside an `if`; thanks for clarifying!